tcms/kiwi_auth/test_admin.py

```diff
@@ -109,3 +109,42 @@ def test_admin_can_add_new_users(self):
 
         self.assertEqual(HTTPStatus.OK, response.status_code)
         self.assertTrue(get_user_model().objects.filter(username='added-by-admin').exists())
+
+
+class TestGroupAdmin(LoggedInTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.admin = UserFactory()
+        cls.admin.is_superuser = True
+        cls.admin.set_password('admin-password')
+        cls.admin.save()
+
+    def test_should_not_be_allowed_to_change_administrator_group_name(self):
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.admin.username,
+            password='admin-password')
+        response = self.client.get('/admin/auth/group/1/change/')
+        self.assertNotContains(response, 'id="id_name"')
+
+    def test_should_not_be_allowed_to_change_tester_group_name(self):
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.admin.username,
+            password='admin-password')
+        response = self.client.get('/admin/auth/group/2/change/')
+        self.assertNotContains(response, 'id="id_name"')
+
+    def test_should_be_allowed_to_create_new_group_and_change_its_name(self):
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.admin.username,
+            password='admin-password')
+        group_name = 'New Group Name'
+        response = self.client.post('/admin/auth/group/add/', {
+            'name': group_name,
+        }, follow=True)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
```

Comment:
adding a new group & editing an existing one (but not Tester & Admin) it should be 2 different scenarios.

Also when adding you are interested in what's on the page and then if the POST will work.